### DISCUSSION TITLE
Play sample one after the other with no silence between

### DISCUSSION THREAD

**Discussion by bsuschi** [naudio/NAudio#990] [Q&A]
I am looking for a way to play samples one after the other.
Sounds simple at first, but the following is needed:

1. It is not known beforehand which sample will be played next. The user decides just before the end of the sample that is being played.
2. The new sample may only be played after the sample currently being played has ended.
3. There must be no silence between samples. A smooth transition is required.
4. In addition, it should be possible to play a sample in a continuous loop until the user clicks on continue.

The samples are processed accordingly. They start and end with a 0 level to prevent cracking.

From what I have read about NAudio so far, I am sure that this can be solved with NAudio.
My attempt via PlaybackStopped results in silence between samples and is not to be used like this.

`wavePlayer.PlaybackStopped += OnPlaybackStopped;`

Can anyone give me some support? At least show me the way?

Thank you and best regards

**Comment by markheath** [comment]
I'd make my own custom ISampleProvider that manages various instances of AudioFileReader to play the samples. In the Read method, when it reaches then end of a sample, it can either loop back to the beginning by resetting the position to 0, or start to play a new one.

**Reply by bsuschi** [reply]
Hello Mark, thanks for your quick reply.
It seems that the problem sitting in front of the screen.
Basically, I understand your answer, but I can't find a start in implementing it. I think my problem is that I haven't really understood Naudio yet. It's probably just a few lines of code.
Can you or anyone else give me a code example?
Thanks a lot

**Reply by markheath** [reply]
Basically if you understand how ISampleProvider works, you can make all kinds of useful components. I'd suggest just looking at a few of the ISampleProvider implementations inside NAudio. The key method is the `Read` method, where you are asked to provide samples. In your case, you'd be reading them from something like an `AudioFileReader` and passing them on.
